packages/api/src/routes/schemas/tenovi-headers.ts

```diff
@@ -0,0 +1,32 @@
+import { Request } from "express";
+import { Context, GetWithoutParams, getFrom } from "../util";
+
+export const tenoviApiKeyPropName = "x-tenovi-api-key";
+export const tenoviClientNamePropName = "x-tenovi-client";
+
+export function getTenoviApiKeyFrom(context: Context, req: Request): GetWithoutParams {
+  return {
+    optional: () => getTenoviHeaderFromOptional(context, tenoviApiKeyPropName, req),
+    orFail: () => getTenoviHeaderFromOrFail(context, tenoviApiKeyPropName, req),
+  };
+}
+
+export function getTenoviClientNameFrom(context: Context, req: Request): GetWithoutParams {
+  return {
+    optional: () => getTenoviHeaderFromOptional(context, tenoviClientNamePropName, req),
+    orFail: () => getTenoviHeaderFromOrFail(context, tenoviClientNamePropName, req),
+  };
+}
```

Comment:
Ideally we wouldn't create these for each param we want to accept. The idea is to have those for very specific cases, and use the code below for everything else:

```ts
getFrom("header").orFail(tenoviApiKeyPropName, req);
```

TBH, I've been wanting to change this pattern for some time, I'm not too fond of how it writes... but since it's working, I'm focusing on other stuff. 😅

Reply:
As discussed offline, I created similar functions that obtain Tenovi-specific information from the headers and throw BadRequestError otherwise.